Publish minutes of 2024-12-05 meeting (#734)
hober/web-extensions: _minutes/2024-12-05-wecg.md

 * [oliver] I expect we're unlikely to break this unless there's a significant issue.
 * [giorgio] If you could document intent to synchronize behavior, I'll stay on the current Firefox behavior (MV2) for now.
 * [oliver] I'll follow up and leave a comment.
 * [rob] Apple folks - what does Safari do here?
 * [timothy] I'll have to test to see what our behavior is in this situation.

[Issue 731](https://github.com/w3c/webextensions/issues/731): Incorrect application of "domainType: thirdParty" in DNR - blocking direct access to iana.org when opening a new tab

 * [dmitry] Title may be inaccurate. We thought the problem was with initiator domain. Five hours ago I found another case where our requests are not blocked. Main issue is with  DNR rules blocking third party requests. We expect that opening a document in a new tab would be a first party request. Seems like a mistake in logic.
 * [timothy] Agree that it sounds like a bug. Not sure what we do, but I know new page loads are treated as first party.
 * [kiara] I just tested, Safari doesn't block with these rules.
 * [rob] Was this observed in Firefox?
 * [simeon] Yes, noted in the issue description.

>  * [oliver] I know we had some discussion about links from 3rd party applications. Might be worth linking. I know the web's concept of first party directly matches DNR's definition or any other concept.

Reviewer: ```suggestion
 * [oliver] I know we had some discussion about links from 3rd party applications. Might be worth linking. I know the web's concept of first party does not directly match DNR's definition or any other concept.
```

Author: Good catch! Meaning was inverted.

>  * [rob] I deferred implementation of domainType in Firefox because of the lack of a clear definition of “first-party”. Ended up mirroring Chrome's implementation, which is modeled after adblocker rules, as described at https://bugzilla.mozilla.org/show_bug.cgi?id=1797408#c6.

Reviewer: ```suggestion
 * [rob] I deferred implementation of `domainType` in Firefox because of the lack of a clear definition of "first-party". Ended up mirroring Chrome's implementation, which is modeled after adblocker rules, as described at https://bugzilla.mozilla.org/show_bug.cgi?id=1797408#c6.
```

[PR 729](https://github.com/w3c/webextensions/pull/729): Add proposal for runtime.onEnabled and runtime.onExtensionLoaded events

 * [sohail] PR for [issue 353](https://github.com/w3c/webextensions/issues/353): Support onEnabled event when the extension from disabled state to enabled state
 * [sohail] Chromium implementation bug: https://crbug.com/1513560
 * [sohail] Formal proposal based on Google's proposal.
 * [oliver] Any notable changes from previous discussion or is it the same
 * [sohail] Same as previous chromium discussion.
 * [rob] Who on Apple's side should be tagged as a reviewer? Oliver on Chrome, me on Firefox. I'll tentatively add Timothy for Apple.
 * [simeon] Anything in particular we should discuss about this today or is it just an FYI?
 * [sohail] Just an FYI.
 * [tomislav] What's the stage internally for Chromium?
 * [oliver] I believe Sohail from Microsoft are proposing to implement this.
 * [mukul] I believe we heard from Chrome that they were open to it but didn't have the bandwidth to do it, so we're taking it on.

[Issue 732](https://github.com/w3c/webextensions/issues/732): Proposal: Add permanentNodeType property to BookmarkTreeNode

>  * [oliver] Looking further into the future, looking at changes to bookmark folders. Before making changes we'd like to look at use cases that are useful to extensions to make it easier to manage bookmarks. Current (Chrome) extension developers look at the ID. There are edge cases where the ID can change. Devs fall back to sting matching. In Firefox the ID isn't numerical. This proposal introduces a new property to identify what are special nodes and what type they are. James and Monica are here from Chrome, they're working on the bookmark changes.

Reviewer: ```suggestion
 * [oliver] Looking further into the future, looking at changes to bookmark folders. Before making changes we'd like to look at use cases that are useful to extensions to make it easier to manage bookmarks. Current (Chrome) extension developers look at the ID. There are edge cases where the ID can change. Devs fall back to string matching. In Firefox the ID isn't numerical. This proposal introduces a new property to identify what are special nodes and what type they are. James and Monica are here from Chrome, they're working on the bookmark changes.
```

>  * [monica] Thanks oliver, no additional notes at the moment.

Reviewer: ```suggestion
 * [monica] Thanks Oliver, no additional notes at the moment.
```

 * [rob] Is there any documentation about the upcoming changes?
 * [oliver] The other issue (733) points to a blog post.
 * [james] Probably a bit early to share beyond what's in the public post.

>  * [monica] To give a general idea, today all bookmarks are local are synced. We're introducing another model where some are local, some are account synced. There may be more than one bookmark bar folder, one for local account and one for synced content.

Reviewer: ```suggestion
 * [monica] To give a general idea, today all bookmarks are local or synced. We're introducing another model where some are local, some are account synced. There may be more than one bookmark bar folder, one for local account and one for synced content.
```

 * [simeon] Is is possible for items to be interposed?
 * [monica] In the extension API they will be separate.
 * [rob] Would permanentNodeType be inherit by bookmarks, e.g. BOOKMARKS_BAR on individual bookmarks, or just the folder at the top level?
 * [monica] The folders are typed, the nodes inside them are normal, so they need to be separate.
 * [rob] As an extension developer I've had to parse the content of a bookmark to figure out it's type, so having an explicit field would be welcome. Oliver, you mentioned extensions parsing the ID - do you have an analysis of use cases to share? I'd like to make sure that what we design matches with actual uses.

>  * [oliver] Nothing off hand, but we can share examples of what we've seen in the wild. It seemed like Chrome, Edge, and Safar had similar concepts for what is synced.

Reviewer: ```suggestion
 * [oliver] Nothing off hand, but we can share examples of what we've seen in the wild. It seemed like Chrome, Edge, and Safari had similar concepts for what is synced.
```

 * [timothy] We don't currently support the Bookmarks API so we'd need to look into it. I don't have details on all of the special cases that we have, but what's been discussed so far sounds good. We should be able to work with it.
 * [tomislav] I think I heard two different things from Oliver and James/Monica. Are we trying to standardize types or are we expecting that they will be browser specific?
 * [monica] This is what we have in Chrome. Oliver?

>  * [oliver] FOr something like the bookmarks bar, it would be nice to be consistent across browsers. Can have additional discussions about what we do for concepts that are more browser specific.

Reviewer: ```suggestion
 * [oliver] For something like the bookmarks bar, it would be nice to be consistent across browsers. Can have additional discussions about what we do for concepts that are more browser specific.
```

 * [simeon] I may have misunderstood. Monica, I thought you said in the future a single folder could contain synced and non-synced items. Is that the intent?
 * [monica] No, content can't be mixed. That's why we have the folder distinction.
 * [timothy] I'm okay with the proposal, but think that it would be nice to not specify the requirement of the syncable state required to inherit from the top folder.
 * [tomislav] Timothy, I think you're suggesting that we shouldn't specify the planned chrome behavior.
 * [timothy] That's what I was originally thinking, but I'm not sure how that would work. Just wanted to be as flexible as possible.
 * [david] I think that makes sense. You might have other bookmarks set by device management that aren't syncable.
 * [oliver] Perhaps an enum? Not sure if that's feasible.
 * [rob] Before we dive into the specifics, we should consider why developers care and whether the design makes sense for their use cases.
 * [monica] If you have two bookmark bar folders, would it be useful to know what the sync behavior difference is between them?

>  * [rob] That sounds useful. On the fly use case, If an extension wants to create a bookmark in the toolbar and make sure they're not synced, how would a developer specify that?

Reviewer: ```suggestion
 * [rob] That sounds useful. On the fly use case, if an extension wants to create a bookmark in the toolbar and make sure they're not synced, how would a developer specify that?
```

>  * [monica] I don't think you'd be able to do so. In general today when you create a bookmark you need to create it in a given folder. When you decide where to put a bookmark, can check whether or not the folder is syncable. Devs should not assume whether or not a folder is syncable.

Reviewer: ```suggestion
 * [monica] I don't think you'd be able to do so. In general today when you create a bookmark you need to create it in a given folder. When you decide where to put a bookmark, you can check whether or not the folder is syncable. Devs should not assume whether or not a folder is syncable.
```

 * [rob] Question: you mentioned no support for register by URL. Content script matching by URL is supported, isn't it?
 * [giorgio] Parameters. Can declare fixed scripts but cannot set parameters for content scripts by tabID or URL.
 * [rob] If you had a choice, would you prefer a way to declare parameters per URL or have sync IPC?
 * [giorgio] Sync IPC would be great, but I don't expect you'd want that. Just need sync messaging once at tab start.
 * [rob] Sync IPC would be unlikely. A more reasonable proposal may be synchronously available storage available in a content script. What kind of quota would you need?
 * [giorgio] If you can't filter by URL and tab, I'm not sure. It depends on the user.
 * [oliver] Global params or per-tab params are definitely still the path we'd like to take. I realize there's been a multi-year delay. Realize adding per-tab is similar in scope to global params, so neither path is obviously easier than the other.
 * [giorgio] 10 MB if there's no filter.

>  * [rob] That's … a lot. From the API design point of view, it would be preferable for extensions to be able to specify different values, so that they are not forced to use 1 giant global “scriptParams”.

Reviewer: ```suggestion
 * [rob] That's … a lot. From the API design point of view, it would be preferable for extensions to be able to specify different values, so that they are not forced to use one giant global "scriptParams".
```

 * [oliver] Does it need to be filtered by the key or the tab?
 * [giorgio] Tab. Need to know in a given tab that you do or don't have to do a particular set of stuff.

>  * [timothy] how do you even know the tab id for the first load? By the time you know the tab ID you're already in the middle of the page load.

Reviewer: ```suggestion
 * [timothy] How do you even know the tab id for the first load? By the time you know the tab ID you're already in the middle of the page load.
```

 * [giorgio] User uses the UI to configure settings for a tab, then reloads the page.
 * [simeon] Does that mean the settings aren't persisted for the URL in that tab?
 * [giorgio] Correct.
 * [timothy] That may work for your implementation, but it doesn't sound sufficient for the general case.
 * [rob] The general use case is to have a default registration independent of tabId, which works across restarts. During a browser session, a user can have the desire for temporary exceptions, for which they can register tab-specific rules/scripts/whatever.
 * [giorgio] DuckDuckGo does something similar. Not just declarative net request.

[Issue 538](https://github.com/w3c/webextensions/issues/538): Proposal: RegisteredContentScript.workers property to inject WorkerScope(s)

>  * (Dropped due to time limitations)

Reviewer: ```suggestion
 * (Skipped due to time limitations)
```
